core/trino-main/src/main/java/io/trino/sql/planner/iterative/rule/PushMergeWriterUpdateIntoConnector.java

```diff
@@ -0,0 +1,148 @@
+        MergeProcessorNode mergeProcessor = captures.get(MERGE_PROCESSOR_NODE_CAPTURE);
+        ProjectNode project = captures.get(PROJECT_NODE_CAPTURE);
+        TableScanNode tableScan = captures.get(TABLE_SCAN);
+
+        List<String> orderedColumnNames = mergeWriter.getTarget().getMergeParadigmAndTypes().getColumnNames();
+        List<? extends Node> mergeAssignments = project.getAssignments().get(mergeProcessor.getMergeRowSymbol()).getChildren();
+
+        ImmutableMap.Builder<String, Constant> assignmentsBuilder = ImmutableMap.builder();
+
+        for (int i = 0; i < orderedColumnNames.size(); i++) {
+            String columnName = orderedColumnNames.get(i);
+            Node assigmentNode = mergeAssignments.get(i);
+            if (!isColumnName(assigmentNode) && !isConstantUpdateValue(assigmentNode)) {
+                // fall back in all cases except constant update values and column names
+                return Result.empty();
+            }
+
+            // if column is unmodified, we just got colum name here,
+            // which we don't want/need to process
+            // we don't support yet any expressions in update statements yet, only constants
+            // also in case of UPDATE ALL statement - row represented as base64 call
+            if (isConstantUpdateValue(assigmentNode)) {
+                ConnectorExpressionTranslator.ConnectorExpressionTranslation expressionTranslation = ConnectorExpressionTranslator.translateConjuncts(
+                        context.getSession(),
+                        ((Expression) assigmentNode),
+                        context.getSymbolAllocator().getTypes(),
+                        plannerContext,
+                        typeAnalyzer);
+
+                ConnectorExpression connectorExpression = expressionTranslation.connectorExpression();
+
+                if (!(connectorExpression instanceof Constant)) {
+                    return Result.empty();
+                }
+                assignmentsBuilder.put(columnName, (Constant) connectorExpression);
+            }
+        }
+        Map<String, Constant> assignments = assignmentsBuilder.buildOrThrow();
+        if (assignments.isEmpty()) {
+            return Result.empty();
+        }
+
+        return metadata.applyUpdate(context.getSession(), tableScan.getTable(), assignments)
+                .map(newHandle -> new TableUpdateNode(
+                        context.getIdAllocator().getNextId(),
+                        newHandle,
+                        getOnlyElement(node.getOutputSymbols())))
+                .map(Result::ofPlanNode)
+                .orElseGet(Result::empty);
+    }
+
+    private static boolean isConstantUpdateValue(Node assigmentNode)
+    {
+        return assigmentNode instanceof Literal || (assigmentNode instanceof Cast && ((Cast) assigmentNode).getExpression() instanceof Literal);
+    }
+
+    private static boolean isColumnName(Node assigmentNode)
```

Comment:
it is not a not column, but symbol. Here we don't know what it represented by this symbol